### DISCUSSION TITLE
Support PostgreSQL parameterized queries

### DISCUSSION THREAD

**Discussion by KazimirPodolski** [mikro-orm/mikro-orm#5595] [Ideas]
**Is your feature request related to a problem? Please describe.**
As far as I see, [Using QueryBuilder to execute native SQL queries](https://mikro-orm.io/docs/6.1/usage-with-sql#using-querybuilder-to-execute-native-sql-queries) is misleading because is does not actually execute the query on the driver, but assumes knex's query format, and only then executes. Actual PostgreSQL's format for parameterized queries is not supported and `execute` fails, expecting only knex's `?` and `:` for parameters, but not native `$`.

**Describe the solution you'd like**
I would like `orm.em.getDriver().execute` to call [`pg`](https://node-postgres.com/features/queries#parameterized-query) instead, or an actual driver, **without knex** as intermediary.

**Describe alternatives you've considered**
Reformatting into knex format is possible, but solves only the half of the problem, as `knex` is still involved.

Implementing [knex-less custom driver](https://mikro-orm.io/docs/6.1/custom-driver) using `pg` directly would solve it too, tho I'm not sure it's actually possible? Would I need to provide another query builder?..

**Additional context**
Motivations:
* I want to integrate [pg-boss](https://github.com/timgit/pg-boss) with mikro-orm so that my jobs are commited/rollbacked along with the "buisiness logic". pg-boss allows integrating with your ORM/driver of choice by [implementing a callback for executing pg-bosses' query](https://github.com/timgit/pg-boss/blob/master/docs/readme.md#newoptions), which is formatted in Postgres native format  (with `$` params)
* knex is known to be [plagued with sql injections](https://evertpot.com/knex-sql-injection/) during it's whole existence. knex reinvents it's own format of parameterized queries and have been failing to properly implement it thought it's whole history. Basically, knex in your project is a built-in unfixable security risk

**Comment by KazimirPodolski** [comment]
I've actually tried different combinations of `execute`, `getDriver().execute`, `getConnection().execute`, `getKnex().raw` and I don't even understand what uses which format.

**Reply by KazimirPodolski** [reply]
Apparently v6 changed that to some direction...

**Comment by B4nan** [comment]
Knex is managing connection pool, you cant just call pg directly and expect things to work fine, its not that simple. In the long run, I would like to add another driver based on kysely, but as of now I am not planning to implement connection pooling in the orm.
